builder/builder.go

```diff
@@ -271,6 +280,16 @@ func (b *Builder) convertAndBuild(ctx context.Context, target domain.Target, opt
 				} else {
 					platform := llbutil.PlatformWithDefault(sts.Platform)
 					platformStr := llbutil.PlatformWithDefaultToString(sts.Platform)
+					platformImgName, err := platformSpecificImageName(saveImage.DockerTag, platform)
+					if err != nil {
+						return nil, err
+					}
+					if platformImgNames[platformImgName] {
```

Comment:
same as above comment.
```suggestion
					if _, found := platformImgNames[platformImgName]; found {
```